### ISSUE TITLE
MacOS catalyst 10.15.1 - Assertion failure in -[_UICollectionLayoutItemSolver _solveGroup]

### ISSUE THREAD

**Issue by nferruzzi** [apptekstudios/ASCollectionView#43]
**Describe the bug**
So basically after I've upgraded from 10.15 to 10.15.1 most Dem layouts just crashes with the exception below (tags, and magazine are working fine).

**To Reproduce**
Run the Demo project enabling Catalyst

**Expected behaviour**
Working like in the sim

**Screenshots**
```
Thread 1 Queue : com.apple.uikit.datasource.diffing (serial)

2019-11-09 11:30:43.798628+0100 ASCollectionViewDemo[13658:253069] *** Assertion failure in -[_UICollectionLayoutItemSolver _solveGroup], /BuildRoot/Library/Caches/com.apple.xbs/Sources/UIKitCore/UIKit-3900.12.4.906/_UICollectionLayoutItemSolver.m:546
2019-11-09 11:30:43.857480+0100 ASCollectionViewDemo[13658:253069] [General] An uncaught exception was raised
2019-11-09 11:30:43.857609+0100 ASCollectionViewDemo[13658:253069] [General] Cannot solve for a non-group item.
2019-11-09 11:30:43.857849+0100 ASCollectionViewDemo[13658:253069] [General] (
	0   CoreFoundation                      0x00007fff37955f53 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6da1b835 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff37971810 +[NSException raise:format:arguments:] + 88
	3   Foundation                          0x00007fff3a0515d1 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 191
	4   UIKitCore                           0x00007fff758e4877 -[_UICollectionLayoutItemSolver _solveGroup] + 3778
	5   UIKitCore                           0x00007fff758e32f0 -[_UICollectionLayoutItemSolver _solveForContainer:layoutAxis:traitCollection:maxFrameCount:layoutRTL:preferredSizes:solutionRecursionDepth:] + 253
	6   UIKitCore                           0x00007fff758e1c48 -[_UICollectionLayoutItemSolver solveForContainer:layoutAxis:traitCollection:maxFrameCount:layoutRTL:preferredSizes:] + 110
	7   UIKitCore                           0x00007fff758e1bbf -[_UICollectionLayoutItemSolver solveForContainer:layoutAxis:traitCollection:maxFrameCount:layoutRTL:] + 77
	8   UIKitCore                           0x00007fff758f571f -[_UICollectionLayoutSectionFixedSolver _solve] + 317
	9   UIKitCore                           0x00007fff758ca213 -[_UICollectionCompositionalLayoutSolver _solveWithSectionLayouts:preferredSizesDict:dataSourceSnapshot:update:] + 1491
	10  UIKitCore                           0x00007fff758c9c29 -[_UICollectionCompositionalLayoutSolver _solve] + 201
	11  UIKitCore                           0x00007fff758c053d -[_UICollectionCompositionalLayoutSolver initWithContainer:traitCollection:layoutAxis:dataSourceSnapshot:options:sectionProvider:] + 659
	12  UIKitCore                           0x00007fff758bc235 -[UICollectionViewCompositionalLayout _fullResolve] + 1216
	13  UIKitCore                           0x00007fff758ba08d -[UICollectionViewCompositionalLayout prepareLayout] + 84
	14  UIKitCore                           0x00007fff755f2dff -[UICollectionViewData _prepareToLoadData] + 297
	15  UIKitCore                           0x00007fff755f1b33 -[UICollectionViewData validateLayoutInRect:] + 53
	16  UIKitCore                           0x00007fff755f1080 -[UICollectionView layoutSubviews] + 240
	17  UIKitCore                           0x00007fff7553b4cc -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 2560
	18  QuartzCore                          0x00007fff430fdba5 -[CALayer layoutSublayers] + 337
	19  QuartzCore                          0x00007fff430fd57f _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 525
	20  UIKitCore                           0x00007fff7553a7cb -[UIView(Hierarchy) layoutBelowIfNeeded] + 1299
	21  UIKitCore                           0x00007fff75525e18 -[UICollectionView reloadData] + 1620
	22  UIKit                               0x00007fff6f0a7e5c -[UICollectionViewAccessibility reloadData] + 42
	23  UIKitCore                           0x00007fff7595d36e -[UICollectionView _performDiffableUpdate:] + 44
	24  UIKitCore                           0x00007fff7591c335 -[_UIDiffableDataSourceViewUpdater _reloadData] + 136
	25  UIKitCore                           0x00007fff7590fc7c __56-[__UIDiffableDataSource reloadFromSnapshot:completion:]_block_invoke + 310
	26  UIKitCore                           0x00007fff7590fcc6 __56-[__UIDiffableDataSource reloadFromSnapshot:completion:]_block_invoke.193 + 31
	27  libdispatch.dylib                   0x000000010072978f _dispatch_client_callout + 8
	28  libdispatch.dylib                   0x000000010073a4cb _dispatch_lane_barrier_sync_invoke_and_complete + 135
	29  UIKitCore                           0x00007fff7590f93b -[__UIDiffableDataSource reloadFromSnapshot:completion:] + 668
	30  UIKitCore                           0x00007fff7590fd92 -[__UIDiffableDataSource applyDifferencesFromSnapshot:animatingDifferences:completion:] + 68
	31  libswiftUIKit.dylib                 0x00007fff78c81a4a $s5UIKit34UICollectionViewDiffableDataSourceC5apply_20animatingDifferences10completionyAA010NSDiffableeF8SnapshotVyxq_G_SbyycSgtFTm + 186
	32  ASCollectionViewDemo                0x00000001000e5518 $s16ASCollectionViewAAV11CoordinatorC18populateDataSource8animatedySb_tF + 1272
	33  ASCollectionViewDemo                0x00000001000e5dd1 $s16ASCollectionViewAAV11CoordinatorC13updateContent_8animated20refreshExistingCellsySo012UICollectionB0C_S2btF + 1441
	34  ASCollectionViewDemo                0x00000001000e052b $s16ASCollectionViewAAV22updateUIViewController_7contextyAA013AS_CollectionbE0C_7SwiftUI0dE20RepresentableContextVyAByxGGtF + 891
	35  ASCollectionViewDemo                0x00000001000ead89 $s16ASCollectionViewAAVyxG7SwiftUI29UIViewControllerRepresentableAadEP06updateeF0_7contexty0eF4TypeQz_AD0efG7ContextVyxGtFTW + 9
	36  SwiftUI                             0x00007fff719ec9d1 $s7SwiftUI42PlatformViewControllerRepresentableAdaptorV06updateD8Provider_7contexty06UIViewE4TypeQz_AA0cdF7ContextVyACyxGGtF + 289
	37  SwiftUI                             0x00007fff71aae81b $s7SwiftUI17PlatformViewChild33_A513612C07DFA438E70B9FA90719B40DLLV6update7contexty14AttributeGraph0O7ContextVyADyxGGz_tFyyXEfU_ + 2331
	38  SwiftUI                             0x00007fff71aa94e6 $s7SwiftUI17PlatformViewChild33_A513612C07DFA438E70B9FA90719B40DLLV6update7contexty14AttributeGraph0O7ContextVyADyxGGz_tF + 310
	39  SwiftUI                             0x00007fff71aafbd0 $s7SwiftUI17PlatformViewChild33_A513612C07DFA438E70B9FA90719B40DLLVyxG14AttributeGraph07UntypedM0AafGP7_update_5graph9attributeySv_So10AGGraphRefaSo11AGAttributeatFZTW + 32
	40  AttributeGraph                      0x00007fff4b548399 $sTA + 25
	41  AttributeGraph                      0x00007fff4b52fa9b _ZN2AG5Graph11UpdateStack6updateEv + 1111
	42  AttributeGraph                      0x00007fff4b52fd59 _ZN2AG5Graph16update_attributeEjb + 377
	43  AttributeGraph                      0x00007fff4b534969 _ZN2AG8Subgraph6updateEj + 929
	44  SwiftUI                             0x00007fff7196b310 $s7SwiftUI9ViewGraphC14runTransaction33_D63C4EB7F2B205694B6515509E76E98BLL2inySo10AGGraphRefa_tF + 224
	45  SwiftUI                             0x00007fff7196b727 $s7SwiftUI9ViewGraphC13updateOutputs2atyAA4TimeV_tFSb5prefs_Sb9idealSizeAC0F0V7outputstSo10AGGraphRefaXEfU_ + 103
	46  SwiftUI                             0x00007fff7196b3e1 $s7SwiftUI9ViewGraphC13updateOutputs2atyAA4TimeV_tF + 145
	47  SwiftUI                             0x00007fff71c516b9 $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayListySd_SbtFyyXEfU_yyXEfU_ + 1001
	48  SwiftUI                             0x00007fff71c510ca $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayListySd_SbtFyyXEfU_ + 634
	49  SwiftUI                             0x00007fff71c46364 $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayListySd_SbtF + 436
	50  SwiftUI                             0x00007fff71dd2752 $s7SwiftUI14_UIHostingViewC14layoutSubviewsyyF + 226
	51  SwiftUI                             0x00007fff71dd2775 $s7SwiftUI14_UIHostingViewC14layoutSubviewsyyFTo + 21
	52  UIKitCore                           0x00007fff7553b4cc -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 2560
	53  QuartzCore                          0x00007fff430fdba5 -[CALayer layoutSublayers] + 337
	54  QuartzCore                          0x00007fff430fd57f _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 525
	55  QuartzCore                          0x00007fff432223fb _ZN2CA7Context18commit_transactionEPNS_11TransactionEd + 323
	56  QuartzCore                          0x00007fff430ec75e _ZN2CA11Transaction6commitEv + 638
	57  QuartzCore                          0x00007fff430ec432 _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 66
	58  CoreFoundation                      0x00007fff3790166e __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	59  CoreFoundation                      0x00007fff37901594 __CFRunLoopDoObservers + 457
	60  CoreFoundation                      0x00007fff378a4852 __CFRunLoopRun + 1514
	61  CoreFoundation                      0x00007fff378a3fe3 CFRunLoopRunSpecific + 499
	62  HIToolbox                           0x00007fff3642b67d RunCurrentEventLoopInMode + 292
	63  HIToolbox                           0x00007fff3642b3bd ReceiveNextEventCommon + 600
	64  HIToolbox                           0x00007fff3642b147 _BlockUntilNextEventMatchingListInModeWithFilter + 64
	65  AppKit                              0x00007fff34ab0864 _DPSNextEvent + 990
	66  AppKit                              0x00007fff34aaf5d4 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
	67  AppKit                              0x00007fff34aa9d76 -[NSApplication run] + 658
	68  AppKit                              0x00007fff34a9b95d NSApplicationMain + 777
	69  AppKit                              0x00007fff34f1c094 _NSApplicationMainWithInfoDictionary + 16
	70  UIKitMacHelper                      0x00007fff686d4731 UINSApplicationMain + 322
	71  UIKitCore                           0x00007fff754c7ba4 UIApplicationMain + 2206
	72  ASCollectionViewDemo                0x000000010002affb main + 75
	73  libdyld.dylib                       0x00007fff6ed7e2e5 start + 1
)
2019-11-09 11:30:43.907697+0100 ASCollectionViewDemo[13658:253069] *** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Cannot solve for a non-group item.'
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff37955f53 __exceptionPreprocess + 250
	1   libobjc.A.dylib                     0x00007fff6da1b835 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff37971810 +[NSException raise:format:arguments:] + 88
	3   Foundation                          0x00007fff3a0515d1 -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 191
	4   UIKitCore                           0x00007fff758e4877 -[_UICollectionLayoutItemSolver _solveGroup] + 3778
	5   UIKitCore                           0x00007fff758e32f0 -[_UICollectionLayoutItemSolver _solveForContainer:layoutAxis:traitCollection:maxFrameCount:layoutRTL:preferredSizes:solutionRecursionDepth:] + 253
	6   UIKitCore                           0x00007fff758e1c48 -[_UICollectionLayoutItemSolver solveForContainer:layoutAxis:traitCollection:maxFrameCount:layoutRTL:preferredSizes:] + 110
	7   UIKitCore                           0x00007fff758e1bbf -[_UICollectionLayoutItemSolver solveForContainer:layoutAxis:traitCollection:maxFrameCount:layoutRTL:] + 77
	8   UIKitCore                           0x00007fff758f571f -[_UICollectionLayoutSectionFixedSolver _solve] + 317
	9   UIKitCore                           0x00007fff758ca213 -[_UICollectionCompositionalLayoutSolver _solveWithSectionLayouts:preferredSizesDict:dataSourceSnapshot:update:] + 1491
	10  UIKitCore                           0x00007fff758c9c29 -[_UICollectionCompositionalLayoutSolver _solve] + 201
	11  UIKitCore                           0x00007fff758c053d -[_UICollectionCompositionalLayoutSolver initWithContainer:traitCollection:layoutAxis:dataSourceSnapshot:options:sectionProvider:] + 659
	12  UIKitCore                           0x00007fff758bc235 -[UICollectionViewCompositionalLayout _fullResolve] + 1216
	13  UIKitCore                           0x00007fff758ba08d -[UICollectionViewCompositionalLayout prepareLayout] + 84
	14  UIKitCore                           0x00007fff755f2dff -[UICollectionViewData _prepareToLoadData] + 297
	15  UIKitCore                           0x00007fff755f1b33 -[UICollectionViewData validateLayoutInRect:] + 53
	16  UIKitCore                           0x00007fff755f1080 -[UICollectionView layoutSubviews] + 240
	17  UIKitCore                           0x00007fff7553b4cc -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 2560
	18  QuartzCore                          0x00007fff430fdba5 -[CALayer layoutSublayers] + 337
	19  QuartzCore                          0x00007fff430fd57f _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 525
	20  UIKitCore                           0x00007fff7553a7cb -[UIView(Hierarchy) layoutBelowIfNeeded] + 1299
	21  UIKitCore                           0x00007fff75525e18 -[UICollectionView reloadData] + 1620
	22  UIKit                               0x00007fff6f0a7e5c -[UICollectionViewAccessibility reloadData] + 42
	23  UIKitCore                           0x00007fff7595d36e -[UICollectionView _performDiffableUpdate:] + 44
	24  UIKitCore                           0x00007fff7591c335 -[_UIDiffableDataSourceViewUpdater _reloadData] + 136
	25  UIKitCore                           0x00007fff7590fc7c __56-[__UIDiffableDataSource reloadFromSnapshot:completion:]_block_invoke + 310
	26  UIKitCore                           0x00007fff7590fcc6 __56-[__UIDiffableDataSource reloadFromSnapshot:completion:]_block_invoke.193 + 31
	27  libdispatch.dylib                   0x000000010072978f _dispatch_client_callout + 8
	28  libdispatch.dylib                   0x000000010073a4cb _dispatch_lane_barrier_sync_invoke_and_complete + 135
	29  UIKitCore                           0x00007fff7590f93b -[__UIDiffableDataSource reloadFromSnapshot:completion:] + 668
	30  UIKitCore                           0x00007fff7590fd92 -[__UIDiffableDataSource applyDifferencesFromSnapshot:animatingDifferences:completion:] + 68
	31  libswiftUIKit.dylib                 0x00007fff78c81a4a $s5UIKit34UICollectionViewDiffableDataSourceC5apply_20animatingDifferences10completionyAA010NSDiffableeF8SnapshotVyxq_G_SbyycSgtFTm + 186
	32  ASCollectionViewDemo                0x00000001000e5518 $s16ASCollectionViewAAV11CoordinatorC18populateDataSource8animatedySb_tF + 1272
	33  ASCollectionViewDemo                0x00000001000e5dd1 $s16ASCollectionViewAAV11CoordinatorC13updateContent_8animated20refreshExistingCellsySo012UICollectionB0C_S2btF + 1441
	34  ASCollectionViewDemo                0x00000001000e052b $s16ASCollectionViewAAV22updateUIViewController_7contextyAA013AS_CollectionbE0C_7SwiftUI0dE20RepresentableContextVyAByxGGtF + 891
	35  ASCollectionViewDemo                0x00000001000ead89 $s16ASCollectionViewAAVyxG7SwiftUI29UIViewControllerRepresentableAadEP06updateeF0_7contexty0eF4TypeQz_AD0efG7ContextVyxGtFTW + 9
	36  SwiftUI                             0x00007fff719ec9d1 $s7SwiftUI42PlatformViewControllerRepresentableAdaptorV06updateD8Provider_7contexty06UIViewE4TypeQz_AA0cdF7ContextVyACyxGGtF + 289
	37  SwiftUI                             0x00007fff71aae81b $s7SwiftUI17PlatformViewChild33_A513612C07DFA438E70B9FA90719B40DLLV6update7contexty14AttributeGraph0O7ContextVyADyxGGz_tFyyXEfU_ + 2331
	38  SwiftUI                             0x00007fff71aa94e6 $s7SwiftUI17PlatformViewChild33_A513612C07DFA438E70B9FA90719B40DLLV6update7contexty14AttributeGraph0O7ContextVyADyxGGz_tF + 310
	39  SwiftUI                             0x00007fff71aafbd0 $s7SwiftUI17PlatformViewChild33_A513612C07DFA438E70B9FA90719B40DLLVyxG14AttributeGraph07UntypedM0AafGP7_update_5graph9attributeySv_So10AGGraphRefaSo11AGAttributeatFZTW + 32
	40  AttributeGraph                      0x00007fff4b548399 $sTA + 25
	41  AttributeGraph                      0x00007fff4b52fa9b _ZN2AG5Graph11UpdateStack6updateEv + 1111
	42  AttributeGraph                      0x00007fff4b52fd59 _ZN2AG5Graph16update_attributeEjb + 377
	43  AttributeGraph                      0x00007fff4b534969 _ZN2AG8Subgraph6updateEj + 929
	44  SwiftUI                             0x00007fff7196b310 $s7SwiftUI9ViewGraphC14runTransaction33_D63C4EB7F2B205694B6515509E76E98BLL2inySo10AGGraphRefa_tF + 224
	45  SwiftUI                             0x00007fff7196b727 $s7SwiftUI9ViewGraphC13updateOutputs2atyAA4TimeV_tFSb5prefs_Sb9idealSizeAC0F0V7outputstSo10AGGraphRefaXEfU_ + 103
	46  SwiftUI                             0x00007fff7196b3e1 $s7SwiftUI9ViewGraphC13updateOutputs2atyAA4TimeV_tF + 145
	47  SwiftUI                             0x00007fff71c516b9 $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayListySd_SbtFyyXEfU_yyXEfU_ + 1001
	48  SwiftUI                             0x00007fff71c510ca $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayListySd_SbtFyyXEfU_ + 634
	49  SwiftUI                             0x00007fff71c46364 $s7SwiftUI16ViewRendererHostPAAE6render8interval17updateDisplayListySd_SbtF + 436
	50  SwiftUI                             0x00007fff71dd2752 $s7SwiftUI14_UIHostingViewC14layoutSubviewsyyF + 226
	51  SwiftUI                             0x00007fff71dd2775 $s7SwiftUI14_UIHostingViewC14layoutSubviewsyyFTo + 21
	52  UIKitCore                           0x00007fff7553b4cc -[UIView(CALayerDelegate) layoutSublayersOfLayer:] + 2560
	53  QuartzCore                          0x00007fff430fdba5 -[CALayer layoutSublayers] + 337
	54  QuartzCore                          0x00007fff430fd57f _ZN2CA5Layer16layout_if_neededEPNS_11TransactionE + 525
	55  QuartzCore                          0x00007fff432223fb _ZN2CA7Context18commit_transactionEPNS_11TransactionEd + 323
	56  QuartzCore                          0x00007fff430ec75e _ZN2CA11Transaction6commitEv + 638
	57  QuartzCore                          0x00007fff430ec432 _ZN2CA11Transaction17observer_callbackEP19__CFRunLoopObservermPv + 66
	58  CoreFoundation                      0x00007fff3790166e __CFRUNLOOP_IS_CALLING_OUT_TO_AN_OBSERVER_CALLBACK_FUNCTION__ + 23
	59  CoreFoundation                      0x00007fff37901594 __CFRunLoopDoObservers + 457
	60  CoreFoundation                      0x00007fff378a4852 __CFRunLoopRun + 1514
	61  CoreFoundation                      0x00007fff378a3fe3 CFRunLoopRunSpecific + 499
	62  HIToolbox                           0x00007fff3642b67d RunCurrentEventLoopInMode + 292
	63  HIToolbox                           0x00007fff3642b3bd ReceiveNextEventCommon + 600
	64  HIToolbox                           0x00007fff3642b147 _BlockUntilNextEventMatchingListInModeWithFilter + 64
	65  AppKit                              0x00007fff34ab0864 _DPSNextEvent + 990
	66  AppKit                              0x00007fff34aaf5d4 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1352
	67  AppKit                              0x00007fff34aa9d76 -[NSApplication run] + 658
	68  AppKit                              0x00007fff34a9b95d NSApplicationMain + 777
	69  AppKit                              0x00007fff34f1c094 _NSApplicationMainWithInfoDictionary + 16
	70  UIKitMacHelper                      0x00007fff686d4731 UINSApplicationMain + 322
	71  UIKitCore                           0x00007fff754c7ba4 UIApplicationMain + 2206
	72  ASCollectionViewDemo                0x000000010002affb main + 75
	73  libdyld.dylib                       0x00007fff6ed7e2e5 start + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Xcode Version:**

XCode 11.2
MacOSX 10.15.1

**Simulator, Device, Both?**
Sim is ok
MacOSX catalyst 10.15 is ok
MacOSX catalyst 10.15.1 is crashing

Well I think it is happening to a lot of projects, see

https://forums.developer.apple.com/thread/124504

so it may not be related to ASCollectionView per se but I wanted to create a bug to keep track of the evolution of this problem with you :) just in case you know any workaround or are already in contact with apple for a fix.

For instance, in my case, I need a simple grid layout with squared items so at the moment I'm using a layout based on the magazine layout because luckily it doesn't crash.

**Comment by apptekstudios** [comment]
I hit the same bug today after upgrading :(

It is most definitely a bug in the catalyst version of UICollectionViewCompositionalLayout. Very unfortunate they let this reach a public release.

The exact same crash is happening with apple's own demo project when you run the ios project on catalyst (https://developer.apple.com/documentation/uikit/views_and_controls/collection_views/using_collection_view_compositional_layouts_and_diffable_data_sources)

**Comment by apptekstudios** [comment]
Fixed in MacOS 10.15.2 🎉
